perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/MailManagerImpl.java

```diff
@@ -884,13 +884,7 @@ private void setFromAndReplyTo(MimeMessage message, String fromName, String from
 
 				// try to use all group admins with specified preferred emails
 				List<RichUser> admins = AuthzResolverBlImpl.getRichAdmins(registrarSession, app.getGroup(), Role.GROUPADMIN);
-				for (User admin : admins) {
-					attrSenderEmail = attrManager.getAttribute(registrarSession, admin, URN_USER_PREFERRED_MAIL);
-
-					if (attrSenderEmail != null && attrSenderEmail.getValue() != null) {
-						result.add(attrSenderEmail.valueAsString());
-					}
-				}
+				result.addAll(getUserPreferredMailsForValidMembers(registrarSession, admins, app.getVo()));
```

Comment:
Groups can have admins assigned through a group from different VO. Therefore, we should check if they are valid in that VO, not in the application's VO. We should also check their status in the group through which they are assigned.